amf-aml/shared/src/main/scala/amf/aml/internal/parse/instances/parser/InstanceNodeParser.scala

```diff
@@ -0,0 +1,160 @@
+package amf.aml.internal.parse.instances.parser
+
+import amf.aml.client.scala.model.domain._
+import amf.aml.internal.metamodel.domain.NodeMappableModel
+import amf.aml.internal.parse.common.AnnotationsParser.parseAnnotations
+import amf.aml.internal.parse.instances.ClosedInstanceNode.checkNode
+import amf.aml.internal.parse.instances.DialectInstanceParser.{computeParsingScheme, emptyElement, typesFrom}
+import amf.aml.internal.parse.instances.InstanceNodeIdHandling.generateNodeId
+import amf.aml.internal.parse.instances.parser.IncludeNodeParser.resolveLink
+import amf.aml.internal.parse.instances.{DialectInstanceContext, NodeMappableHelper}
+import amf.aml.internal.validate.DialectValidations.DialectError
+import amf.core.client.scala.model.domain.DomainElement
+import amf.core.client.scala.parse.document.SyamlParsedDocument
+import amf.core.internal.parser.domain.Annotations
+import amf.core.internal.parser.{Root, YMapOps}
+import org.yaml.model._
+
+case class InstanceNodeParser(root: Root) extends NodeMappableHelper {
```

Comment:
Can move all implicit DialectInstanceContext method parameters to class parameter

Reply:
I didn't put it as class parameter because the intent is for this class to actually have only static methods. I had to add the "root" parameter because it was too cumbersome to edit all of the parsers method signatures to propagate the Root object.